rwiki/rwiki-impl/impl/src/java/uk/ac/cam/caret/sakai/rwiki/component/service/impl/RWikiObjectServiceImpl.java

```diff
@@ -326,6 +331,44 @@ public RWikiCurrentObject getRWikiObject(String name, String realm,
 				}
 				return returnable;
 			}
+			else if(returnable.getGroupsAsArray() != null)
+			{
+				Site site = null;
+				String siteId = wikiSecurityService.getSiteId();
+				try {
+					site = siteService.getSite(siteId);
+				} catch (IdUnusedException idEx) {
+					log.warn("Site not found for id: {}", siteId);
+				}
+				String[] groups = returnable.getGroupsAsArray();
+				String owner = returnable.getOwner();
+				User currentUser = null;
+				try {
+					currentUser = userDirectoryService.getUser(user);
+				} catch(UserNotDefinedException userEx) {
+					log.warn("User not found for id: {}", user);
```

Comment:
Why are we checking if the user exists? I'm asking because it seems we don't really do anything if that's not the case... Not sure what's missing but this verification is useless otherwise.

Reply:
Ok. Sorry. Thanks

Reply:
Nothing to sorry about! You've done a great work, these are just minor things to improve